queries/places.py

```diff
@@ -0,0 +1,352 @@
+    response = dict(answer=answer)
+
+    return response, answer, voice
+
+
+def answ_openhours(placename, loc, qtype):
+    # Look up placename in places API
+    res = query_places_api(
+        placename,
+        userloc=loc,
+        fields="opening_hours,place_id,formatted_address,geometry",
+    )
+    if res["status"] != "OK" or "candidates" not in res or not res["candidates"]:
+        return gen_answer(_PLACES_API_ERRMSG)
+
+    from pprint import pprint
+    pprint(res)
+
+    # Use top result
+    place = res["candidates"][0]
+    if not "opening_hours" in place:
+        return gen_answer("Ekki tókst að sækja opnunartíma fyrir " + icequote(placename))
+
+    place_id = place["place_id"]
+    is_open = place["opening_hours"]["open_now"]
+    # needs_disambig = len(res["candidates"]) > 1
+    fmt_addr = place["formatted_address"]
+
+    # Make sure it's in Iceland
+    coords = _parse_coords(place)
+    if not coords or not in_iceland(coords):
+        return gen_answer(_NOT_ICELAND_ERRMSG)
+
+    # Look up place ID in Place Details API to get more information
+    res = query_place_details(place_id, fields="opening_hours,name")
+    if res["status"] != "OK" or not res or "result" not in res:
+        return gen_answer(_PLACES_API_ERRMSG)
+
+    pprint(res)
+
+    now = datetime.utcnow()
+    # Sun is index 0, as req. by Google API
+    wday = int(now.strftime("%w"))
+
+    try:
+        name = res["result"]["name"]
+        # Generate placename w. street, e.g. "Forréttabarinn á Nýlendugötu"
+        street = fmt_addr.split()[0].rstrip(",")
+        street_þgf = "{nl:þgf}".format(nl=NounPhrase(street))
+        name = "{0} {1} {2}".format(name, iceprep_for_street(street), street_þgf)
+
+        # Get opening hours for current weekday
+        periods = res["result"]["opening_hours"]["periods"]
+        if len(periods) == 1 or wday > len(periods)-1:
+            # Open 24 hours a day
+            today_desc = p_desc = "{0} er opin allan sólarhringinn".format(name)
+        else:
+            p = periods[wday]
+            opens = p["open"]["time"]
+            closes = p["close"]["time"]
+            # Format correctly
+            openstr = opens[:2] + ":" + opens[2:]
+            closestr = closes[:2] + ":" + opens[2:]
+            p_desc = "{0} - {1}".format(openstr, closestr)
+            p_voice = p_desc.replace("-", "til")
+            # TODO: opin vs. opinn vs. opið
+            today_desc = "Í dag er {0} opin frá {1}".format(name, p_voice)
+    except Exception as e:
+        logging.warning("Exception generating answer for opening hours: {0}".format(e))
+        return gen_answer(_PLACES_API_ERRMSG)
+
+    # Generate answer
+    if qtype == "OpeningHours":
+        answer = p_desc
+        voice = today_desc
+    # Is X open? Is X closed?
+    elif qtype == "IsOpen" or qtype == "IsClosed":
+        yes_no = (
+            "Já"
+            if (is_open and qtype == "IsOpen" or not is_open and qtype == "IsClosed")
```

Comment:
I think you need this, at least for clarity:

`if ((is_open and qtype == "IsOpen") or (not is_open and qtype == "IsClosed"))`